src/base/static/components/organisms/dashboard/fixed-table.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+
+import makeParsedExpression from "../../../utils/expression/parse";
+import BaseTable from "./base-table";
+
+const fixedTablePropTypes = {
+  data: PropTypes.shape({
+    columns: PropTypes.arrayOf(
+      PropTypes.shape({
+        dataKey: PropTypes.string.isRequired,
+        label: PropTypes.string.isRequired,
+        type: PropTypes.string.isRequired,
+        fractionalWidth: PropTypes.number.isRequired,
+      }).isRequired,
+    ).isRequired,
+    rows: PropTypes.arrayOf(
+      PropTypes.arrayOf(
+        PropTypes.shape({
+          value: PropTypes.array.isRequired,
+          label: PropTypes.string,
+          type: PropTypes.string.isRequired,
+        }).isRequired,
+      ).isRequired,
+    ).isRequired,
+    headers: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+  }).isRequired,
+  header: PropTypes.string.isRequired,
+  layout: PropTypes.shape({
+    start: PropTypes.number.isRequired,
+    end: PropTypes.number.isRequired,
+    height: PropTypes.string,
+  }).isRequired,
+  stripeColor: PropTypes.string,
+};
+
+type Props = PropTypes.InferProps<typeof fixedTablePropTypes>;
+
+const getFixedTableData = ({ places, widget, widgetState }) => {
```

Comment:
Is there a reason why we didn't add type definitions to these function arguments? It seems like we can really benefit from them here.

Reply:
Sounds good-- definitely feel free to point out any holes in the typings that you see. Knowing how extensively to type things is probably the biggest unknown for me still.